### ISSUE TITLE
Kind multi-cluster setup

### ISSUE THREAD

**Issue by versuchs-kaninchen** [kubernetes-sigs/kind#2018]
Hi all,

I had a question on how to setup connectivity for the API Servers of two different kind clusters?
```
 cluster1                            cluster2
+------------------+                +-----------------+
|                  |                |                 |
|    +-------+  <----------------------+ +--------+   |
|    |  API  |     |                |    |  API   |   |
|    |       | +-----------------------> |        |   |
|    +-------+     |                |    +--------+   |
+------------------+                +-----------------+
```

I am using Istio (dw this is not important) that configures the clusters to reach each other through localhost such as `https://127.0.0.1:49288` but understandably the connection refuses as this is querying the localhost of the kind node.

How can I get this out of the kind node and into my computers localhost?

I am suspecting it will work by configuring the networking: https://kind.sigs.k8s.io/docs/user/configuration/#networking
But was not sure to what should I set it?

Help is very needed
Thanks!

**Comment by BenTheElder** [comment]
The API servers are all on the `kind` network, so they have connectivity.

If you want to connect to the API server from within the clusters / docker, you need to use the kubeconfig from `kind get kubeconfig --internal`, this is because in order to expose the apiserver to the host we use a docker port forward, and because `127.0.0.1` is local to each container as you mentioned.

the node components and workloads similarly use this docker network internal address instead of the host kubeconfig, typically using the automatic pod credentials or a kubernetes service account, rather than using the host KUBECONFIG:

TLDR: 
- Use `node-name:6443` for the `server` instead of the loopback used by the host config
  - You can get this with `kind get kubeconfig --internal`
  - please don't hardcode this though, as e.g. for multiple control plane nodes it's actually a loadbalancer node instead
- It's not possible for us to make the address identical from the host or not (due to docker networking), but the in-cluster credentials from one cluster are what you need to reach it from another

**Comment by BenTheElder** [comment]
also to be explicit: you don't need to make any changes to kind, the application(s) just need to use the correct details for contacting the server from within docker rather than the host.

ideally applications should support this because it's analogous to having a cluster in a private network / data center with an external address for the API via a jump-box / loadbalancer / ... and an internal endpoint within the network (the network here is docker instead and containers instead of datacenter boxes)

I think this is answered so i'm going to close it to keep the tracker tidy, but I'll continue responding to further comments and re-open if we're missing something 😅
